icu4j/main/classes/core/src/com/ibm/icu/message2/FormattedPlaceholder.java

```diff
@@ -0,0 +1,76 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.message2;
+
+import com.ibm.icu.text.FormattedValue;
+
+/**
+ * An immutable, richer formatting result, encapsulating a {@link FormattedValue},
+ * the original value to format, and we are considering adding some more info.
+ * Very preliminary.
+ *
+ * @internal ICU 72 technology preview
+ * @deprecated This API is for ICU internal use only.
+ */
+@Deprecated
+public class FormattedPlaceholder {
+    private final FormattedValue formattedValue;
+    private final Object inputValue;
+
+    /**
+     * Constructor creating the {@code FormattedPlaceholder}.
+     *
+     * @param formattedValue the result of formatting the placeholder.
+     * @param inputValue the original value to be formatted.
```

Comment:
swap docs like params
```suggestion
     * @param inputValue the original value to be formatted.
     * @param formattedValue the result of formatting the placeholder.
```